### PR TITLE
[#2892] add migration for dropping the dx in a doi

### PR DESCRIPTION
**Pull request by sblack-usu**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Find a resource with an existing doi that begins with 'http://dx.'.  Run manage.py migration.  Check the doi and verify it begins with 'https://doi'

### CODE REVIEW

File: hs_core/migrations/0038_doi_url.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from hs_core.models import Identifier
+
+from django.db import migrations
+
+
+def remove_dx(apps, schema_editor):
+    for id in Identifier.objects.filter(url__startswith='http://dx.'):
+        id.url = id.url.replace('http://dx.', 'https://')
```
Comment (aphelionz) [inline hs_core/migrations/0038_doi_url.py]: Do we have any resources that start with `https://dx.`?

File: hs_core/tests/test_dx_migration.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        )
+        resource.save()
+        id = Identifier.objects.all().first()
+        id.url = "http://dx.doi.org/123"
+        id.save()
+
+
+    def test_delimiter_migration(self):
+        id = Identifier.objects.all().first()
+        self.assertEqual('https://doi.org/123', id.url)
```
Comment (aphelionz) [inline hs_core/tests/test_dx_migration.py]: This is cool

File: hs_core/migrations/0038_doi_url.py (hunk 1 above)
Comment (sblack-usu) [reply hs_core/migrations/0038_doi_url.py]: no, every doi was hardocded to `http://dx.`.  https://github.com/hydroshare/hydroshare/commit/8f0a5f2815c47278044a2382053278bd49557407 shows that @katrinleinweber put in some work months ago to change new DOIs to `https://` which was merged to develop but never pushed out.  

I also ran a filter on `https://dx.` just to see if there were any matches and I did not find any.

**Comment by sblack-usu** [comment]
retest this, please

File: hs_core/migrations/0038_doi_url.py (hunk 1 above)
Comment (katrinleinweber) [reply hs_core/migrations/0038_doi_url.py]: I used [`https?:\/\/(dx)?\.doi\.org`](https://regexper.com/#https%3F%3A\%2F\%2F(dx\.)%3Fdoi\.org\%2F) to catch both variants back then ;-)

**Comment by aphelionz** [comment]
@sblack-usu merge when ready

**Pull request merged**